src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionValidator.cs

```diff
@@ -205,6 +205,18 @@ private SelectionInfo GetInitialSelectionInfo(SyntaxNode root, SourceText text)
                 };
             }
 
+            var commonRootContainer = commonRoot.FirstAncestorOrSelf<SyntaxNode>(node => node is ClassDeclarationSyntax or RecordDeclarationSyntax or StructDeclarationSyntax);
+            if (commonRootContainer is null)
+            {
+                return new SelectionInfo
+                {
+                    Status = new OperationStatus(OperationStatusFlag.None, CSharpFeaturesResources.No_valid_selection_to_perform_extraction),
```

Comment:
i think we might want to condier a better string message here.  SPecifically "Selection was not contained inside a type"

also, do we ahve tests for extracing from local functions inside the root namespace?  we don't want to break that (if that works today. ) if it doesn't work prior to your change, tehn no worries.

Reply:
> "Selection was not contained inside a type" 

This may not be a problem for top level statements right?

> "Also, do we ahve tests for extracing from local functions inside the root namespace?"

Is that valid?

Reply:
yup.  top level functions are a thing :)

Reply:
top level functions seem to have been added in C# 9? How can I test if it works?

Reply:
There are no tests for extracting from the root namespace, and I tested it and it has the same error as the bug linked.